.travis.yml

```diff
@@ -2,11 +2,24 @@ language: python
 python:
 - '3.6'
 dist: trusty
-# We're going to ignore all branches except:
-branches:
-  only:
-  - master
-  - production
+
+# Deploy only if:
+#     - it's regular push to the master or production branches
+#     - It's a pull request and:
+#       - PR_AUTOBUILD env. variable is set to "true" or
+#       - the name of the PR branch (head name) is contained in PR_AUTODEPLOY env. variable or
+#       - git commit message contains "[joplin:build]"
+#
+#   Note: PR_AUTOBUILD or PR_AUTODEPLOY variables can be set either in Travis > Settings or
+#   at the env: global: section of this configuration file. See example below.
+#
+
+if: |
+  (branch IN (master, production) AND type = push) OR \
+  (type = pull_request AND env(PR_AUTOBUILD) IN (true, enabled, on)) OR \
+  (type = pull_request AND env(PR_AUTODEPLOY) =~ concat(head_branch) ) OR \
+  (type = pull_request AND commit_message =~ /\[joplin\:build\]/ )
```

Comment:
Do we want this override? I think it makes more sense to remove it.

Reply:
Do you mean the "[joplin:build]" keyword? I personally like the convenience it gives me to build without having to change code or changing environment variables, especially if I only need one quick build. 

It would be very difficult to type that by accident, and it would only work for PRs.

Reply:
Got it, I just set up the new logic:
(branch IN (master, production) AND type = push) OR (type = pull_request)

Meaning it will build all updates to PRs, master and production branches, leaving regular branches alone.